tests/circular-queue.spec.ts

```diff
@@ -0,0 +1,801 @@
+	beforeAll(() => {
+		instance = new ArmorCircularQueue<number>(maxSize);
+	});
+
+	beforeEach(() => {
+		instance.clear();
+		instance.overwrite = false;
+	});
+
+	describe('constructor', () => {
+		it('should initialize with maxSize if it is an integer > 0', () => {
+			const custom1 = new ArmorCircularQueue<number>(1);
+			expect(custom1.state.maxSize).toBe(1);
+			const custom20 = new ArmorCircularQueue<number>(20);
+			expect(custom20.state.maxSize).toBe(20);
+		});
+
+		it('should initialize with maxSize of 256 if maxSize is <= 0', () => {
+			const custom1 = new ArmorCircularQueue<number>(-1);
+			const custom2 = new ArmorCircularQueue<number>(0);
+			expect(custom1.state.maxSize).toBe(256);
+			expect(custom2.state.maxSize).toBe(256);
+		});
+
+		it('should initialize with state when passed one', () => {
+			const state: ArmorCircularQueueState<number> = {
+				elements: items,
+				type: 'cqState',
+				maxSize: 9,
+				size: 2,
+				front: 3,
+				rear: 5
+			};
+			const custom = new ArmorCircularQueue<number>(maxSize, {state: state});
+
+			expect(JSON.parse(custom.stringify()!)).toStrictEqual(state);
+		});
+	});
+
+	describe('isInteger', () => {
+		it('should return true if n is an integer', () => {
+			const types = [-9, -1, 0, 1, 9];
+			types.forEach((type) => {
+				expect(instance.isInteger(type)).toBe(true);
+			});
+		});
+
+		it('should return false if n is not an integer', () => {
+			const types = [-9.9, -1.5, -0.5, 0.5, 1.5, 9.5, null, undefined];
+			types.forEach((type) => {
+				expect(instance.isInteger(type!)).toBe(false);
+			});
+		});
+	});
+
+	describe('isValidState', () => {
+		it('should return true if state is a valid ArmorCircularQueueState', () => {
+			const custom = new ArmorCircularQueue<number>(maxSize);
+			expect(custom.isValidState(custom.state)).toBe(true);
+		});
+
+		it('should return false if state is null or undefined', () => {
+			const custom = new ArmorCircularQueue<number>(maxSize);
+			const types = [null, undefined];
+			types.forEach((type) => {
+				custom.state = type!;
+				expect(custom.isValidState(custom.state)).toBe(false);
+			});
+		});
+
+		it('should return false if state.type is not "cqState"', () => {
+			const custom = new ArmorCircularQueue<number>(maxSize);
+			const types = [null, undefined, ''];
+			types.forEach((type) => {
+				custom.state.type = type as 'cqState';
+				expect(custom.isValidState(custom.state)).toBe(false);
+			});
+		});
+
+		it('should return false if state.size is not an integer >= 0', () => {
+			const custom = new ArmorCircularQueue<number>(maxSize);
+			const types = [-9, -1.5, -1, 0.5, 10.5, null, undefined];
+			types.forEach((type) => {
+				custom.state.size = type!;
+				expect(custom.isValidState(custom.state)).toBe(false);
+			});
+		});
+
+		it('should return false if state.maxSize is not an integer >= 1', () => {
+			const custom = new ArmorCircularQueue<number>(maxSize);
+			const types = [-9, -1.5, -1, 0, 0.5, 10.5, null, undefined];
```

Comment:
Same additions to test types as above comment. Probably easier to add a constant at top of tests called 
```const INVALID_INT_VALUES = [];```
and use that for each test where testing values which are not integers.